dashboard/config/routes.rb

```diff
@@ -417,6 +417,9 @@ module OPS
   get '/dashboardapi/v1/pd/k5workshops', to: 'api/v1/pd/workshops#k5_public_map_index'
   get '/api/v1/pd/workshops_user_enrolled_in', to: 'api/v1/pd/workshops#workshops_user_enrolled_in'
 
+  post '/api/v1/sections/:section_code/join', action: 'join', controller: 'api/v1/sections', defaults: {format: 'json'}
+  post '/api/v1/sections/:section_code/leave', action: 'leave', controller: 'api/v1/sections', defaults: {format: 'json'}
```

Comment:
You're going to have to merge your new `Api::V1::SectionsController` with the one Josh and I recently added.  When you do, you should move these routes into the new `:section_api_routes` concern.  They'll be defined as member actions something like this:

```ruby
  # Section API routes (JSON only)
  concern :section_api_routes do
    defaults format: :json do
      resources :sections, only: [:index, :show, :create] do
        resources :students, only: [:index], controller: 'sections_students'
        member do
          get 'join'
          get 'leave'
        end
      end
    end
  end
```